minibench/tests/test_samplers.py

```diff
@@ -0,0 +1,161 @@
+            fpath, slice_shape, mmap_mode='r', max_count=max_count)
+
+        for rand_slice in sampler:
+            assert rand_slice.shape == slice_shape
+
+        # Get a few of a different shape
+        slice_shape = (1, 20)
+        sampler = minibench.samplers.one_npy_random_slice(
+            fpath, slice_shape, mmap_mode='r', max_count=max_count)
+
+        for rand_slice in sampler:
+            assert rand_slice.shape == slice_shape
+
+
+def test_one_npz_load_random_slice(npz_files):
+    max_count = 3
+    field = 'data'
+    for fpath in npz_files:
+        # Get a few of one kind of shape
+        slice_shape = (3, 2)
+        slicer = minibench.samplers.one_npz_random_slice(
+            fpath, field, slice_shape, max_count=max_count)
+
+        for rand_slice in slicer:
+            assert rand_slice.shape == slice_shape
+
+        # Get a few of a different shape
+        slice_shape = (1, 20)
+        slicer = minibench.samplers.one_npz_random_slice(
+            fpath, field, slice_shape, max_count=max_count)
+
+        for rand_slice in slicer:
+            assert rand_slice.shape == slice_shape
+
+
+def test_one_h5py_random_slice(h5py_file):
+    max_count = 3
+
+    fp = h5py.File(h5py_file)
+    for key in fp:
+        # Get a few of one kind of shape
+        slice_shape = (3, 2)
+        sampler = minibench.samplers.one_h5py_random_slice(
+            key, fp, slice_shape, max_count=max_count)
+
+        for rand_slice in sampler:
+            assert rand_slice.shape == slice_shape
+
+        # Get a few of a different shape
+        slice_shape = (1, 20)
+        sampler = minibench.samplers.one_h5py_random_slice(
+            key, fp, slice_shape, max_count=max_count)
+
+        for rand_slice in sampler:
+            assert rand_slice.shape == slice_shape
+
+
+def test_one_biggie_random_slice(stash_file):
+    max_count = 3
+    field = 'data'
+
+    stash = biggie.Stash(stash_file)
+    for key in stash.keys():
+        # Get a few of one kind of shape
+        slice_shape = (3, 2)
+        sampler = minibench.samplers.one_biggie_random_slice(
+            key, stash, field, slice_shape, max_count=max_count)
+
+        for rand_slice in sampler:
+            assert rand_slice.shape == slice_shape
+
+        # Get a few of a different shape
+        slice_shape = (1, 20)
+        sampler = minibench.samplers.one_biggie_random_slice(
+            key, stash, field, slice_shape, max_count=max_count)
+
+        for rand_slice in sampler:
+            assert rand_slice.shape == slice_shape
+
+
+def test_mux_random_slice(npy_files):
+    slice_shape = (3, 2)
+    sampler = minibench.samplers.mux_random_slice(
+        sampler=minibench.samplers.one_npy_random_slice,
+        collec=npy_files, shape=slice_shape, mmap_mode='r',
+        max_count=3, with_replacement=False)
+
+    # Run the sampler to exhaustion.
+    for rand_slice in sampler:
+        assert rand_slice.shape == slice_shape
```

Comment:
should only have to test one ... and actually, this should probably be tested with a lambda / pass through function, but since `one_npy_random_slice` is already tested inline, I think it's fine.